components/flow/flow-service/src/main/kotlin/net/corda/flow/fiber/cache/impl/FlowFiberCacheImpl.kt

```
@@ -71,10 +71,21 @@ class FlowFiberCacheImpl @Activate constructor(
     }
 
     override fun put(key: FlowKey, suspendCount: Int, fiber: FlowFiber) {
+        if (Thread.currentThread().isInterrupted) {
+            val msg = "Interrupted thread prevented from writing into flow fiber cache with flow key $key"
+            logger.warn(msg)
+            throw InterruptedException(msg)
+        }
+        logger.info("Putting fiber into cache with key $key and suspend count $suspendCount")
         cache.put(key, FiberCacheValue(fiber, suspendCount))
     }
 
     override fun get(key: FlowKey, suspendCount: Int, sandboxGroupId: UUID): FlowFiber? {
+        if (Thread.currentThread().isInterrupted) {
+            val msg = "Interrupted thread prevented from getting from flow fiber cache for key $key suspendCount $suspendCount"
+            logger.warn(msg)
+            throw InterruptedException(msg)
```

this protects from edge case where two threads could try get the key at the same suspend point, to protect against the interrupted thread getting the fiber and mutating it

---

One thing I'm suspect of - it knows its interrupted here, throws the exception, and caller of this (`FlowFiberFactoryImpl`) swallows the exception and deserializes the fiber from checkpoint. Feels like we should throw InterruptedException at the caller